src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -39,5 +33,19 @@ object HydraArtifactsCache {
     }
   }
 
-  private def env(name: String): Option[String] = Option(System.getenv(name))
+  private def cacheArtifacts(artifacts: String, scalaVersion: String, hydraVersion: String) = {
+    val paths = artifacts.split("\n").filter(s => ArtifactsRegex.findFirstIn(s).nonEmpty).map(s => SplitRegex.split(s)(1))
+    val hydraBridge = s"${findDownloadFolder(paths.head)}$GroupId" / "hydra-bridge_1_0" / "srcs" / s"hydra-bridge_1_0-$hydraVersion-sources.jar"
+    val artifactPaths = hydraBridge +: paths
+    cache.put((scalaVersion, hydraVersion), artifactPaths)
+    artifactPaths
+  }
+
+  private def findDownloadFolder(path: String) = {
+    path.split(GroupId).head
+  }
+
+  private implicit class Path(path: String) {
```

Comment:
This can be declared as a value class (less allocation) as follows

```
private implicit class Path(val path: String) extends AnyVal {
  ...
}
```

Reply:
Actually, there are 3 or 4 `RichFile` implicit classes already in the project, maybe you can reuse one of them:

- `org.jetbrains.sbt.RichFile`
- `org.jetbrains.plugins.scala.project.template.FileExt`